### DISCUSSION TITLE
More fine-grained control when syncing fork branch via API

### DISCUSSION THREAD

**Discussion by katexochen** [community/community#61101] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

Hey, it would be super cool if the GitHub REST API would allow more fine-grained control over[ fork syncing](https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#sync-a-fork-branch-with-the-upstream-repository), especially regarding the merge-type. Right now, the response will tell if a branch was merged or fast-forwarded: 

```
// Response
 "merge_type": {
      "type": "string",
      "enum": [
        "merge",
        "fast-forward",
        "none"
      ]
    },
```

In some situation, a user might only ever want to fast-forward their fork, but never create a merge commit. This isn't currently possible, and cleanup after an unwanted merge isn't easy.

**Comment by Sheikhlipu123** [comment]
I believe that the GitHub REST API would be more powerful and flexible if it allowed more fine-grained control over fork syncing. This would make it easier for users to manage their forks and avoid unwanted merges.

Here are some specific suggestions for how the GitHub REST API could be improved:

Add a fast-forward option to the merge_type parameter in the update-branch API endpoint.
Add a new API endpoint that allows users to force-fast-forward a branch.
Add a new API endpoint that allows users to list all of the merge commits for a branch.

**Reply by katexochen** [reply]
> Add a fast-forward option to the merge_type parameter in the update-branch API endpoint.

Yes, this would be my preferred solution. But notice that `merge_type` parameter is currently only present in the response. It would be great if it could be added to the request.
